cohydra/interface.py

```diff
@@ -149,10 +151,18 @@ def connect_tap_to_bridge(self, bridge_name=None):
         # ConfigureLocal is used to prevent the TAP / bridged device to use a "learned" MAC address.
         # So, we can set the CSMA and WiFiNetDevice address to something we control.
         # Otherwise, WiFi ACK misses happen.
-        tap_helper.SetAttribute('Mode', core.StringValue('ConfigureLocal'))
-        tap_helper.SetAttribute('DeviceName', core.StringValue(self.tap_name))
-        tap_helper.SetAttribute('MacAddress', ns_net.Mac48AddressValue(ns_net.Mac48Address.Allocate()))
-        tap_helper.Install(self.node.ns3_node, self.ns3_device)
+        if tap_mode == "ConfigureLocal":
+            tap_helper.SetAttribute('Mode', core.StringValue('ConfigureLocal'))
+            tap_helper.SetAttribute('DeviceName', core.StringValue(self.tap_name))
+            tap_helper.SetAttribute('MacAddress', ns_net.Mac48AddressValue(ns_net.Mac48Address.Allocate()))
+            tap_helper.Install(self.node.ns3_node, self.ns3_device)
+        elif tap_mode == "UseLocal":
+            tap_helper.SetAttribute("Mode", core.StringValue("UseLocal"))
+            tap_helper.SetAttribute("DeviceName", core.StringValue(self.tap_name))
+            tap_helper.Install(self.node.ns3_node, self.ns3_device)
+        else:
+            logger.error("Unsupported TAP-Mode %s.", tap_mode)
```

Comment:
Is it useful to continue execution after running into this branch? Or should we (raise an error and) abort?